Downloads improvement

TcT2k opened the pull request:
A few improvements for the downloads page.

First the downloads page releases are now generated from data files which should make updating to a new version much easier. The JSON data files (`_data/release_assets/`) will be downloaded with the `update_release_asset_info` script. `_data/releases.yml` defines which versions are shown.
Download URLs, file sizes etc, are automatically generated at build time.

Additionally I've integrated the wxMSW DLL Downloads which should make getting started with them much easier, the downloads are grouped by compiler.

I've attached screenshots on how it looks now:

![image](https://user-images.githubusercontent.com/5075894/46734482-9fea6e80-cc93-11e8-9a5e-b4210c1eb808.png)

The button will open the this modal:
![image](https://user-images.githubusercontent.com/5075894/46755527-9417a000-ccc5-11e8-9077-894724c1059a.png)

vadz commented on downloads/index.md:
```diff
@@ -1,6 +1,53 @@
+  - description: "Windows ZIP"
+    prefix: "wxWidgets-"
+    postfix: ".zip"
+  - description: "Windows 7Z"
```
This is _really_ nitpicking, but I think `7z` is much more usual, i.e. "z" shouldn't be capitalized.

vadz commented on downloads/index.md:
```diff
@@ -1,6 +1,53 @@
+  - description: "Windows Installer"
+    prefix: "wxMSW-"
+    postfix: "-Setup.exe"
+  - description: "Source for Linux, OS X, etc"
```
Mmm, shouldn't be this be "macOS" as per the previous PR?

vadz commented on downloads/index.md:
```diff
@@ -1,6 +1,53 @@
+  - description: "Manual (CHM)"
+    postfix: "-docs-chm.zip"
+compilers:
+  - description: "Visual Studio 2008"
```
Perhaps it'd better to use reverse chronological order here so that the most recent compiler is at the top?

TcT2k commented:
@vadz I've updated this to include your comments and updated changed the update script to ruby which makes the update process even easier.
I'll create a PR with updated instructions in the main repo

tierra commented:
Looks good to me, great idea pulling release info from GitHub API.

vadz commented:
Thanks again Tobias!
